### PR TITLE
refactoring timeouts Selenium

### PR DESCRIPTION
**Pull request by jnioche**
see #1101

### CODE REVIEW

File: core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java [hunk 1]
```diff
@@ -72,13 +72,20 @@ public void configure(Config conf) {
             try {
                 RemoteWebDriver driver =
                         new RemoteWebDriver(new URL(cdaddress), capabilities, tracing);
+                // setting timouts
+                // see https://www.browserstack.com/guide/understanding-selenium-timeouts
                 Timeouts touts = driver.manage().timeouts();
-                int implicitWait = ConfUtils.getInt(conf, "selenium.implicitlyWait", 0);
```
Comment (rzo1) [inline core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java:76]: Shall we add a (temporary) logging for these old settings for a smooth transision?

Something like

```
selenium.implicitlyWait is deprecated. Please use selenium.timeouts!
```

and fail hard?

File: core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java [hunk 2]
```diff
@@ -72,13 +72,20 @@ public void configure(Config conf) {
...
-                int scriptTimeout = ConfUtils.getInt(conf, "selenium.scriptTimeout", 0);
-                touts.implicitlyWait(Duration.ofMillis(implicitWait));
-                touts.pageLoadTimeout(Duration.ofMillis(pageLoadTimeout));
-                touts.scriptTimeout(Duration.ofMillis(scriptTimeout));
+                Map<String, Number> timeouts = (Map<String, Number>) conf.get("selenium.timeouts");
+                if (timeouts != null) {
+                    long implicitTimeout = timeouts.getOrDefault("implicit", -1).longValue();
+                    long pageLoadTimeout = timeouts.getOrDefault("pageLoad", -1).longValue();
+                    long scriptTimeout = timeouts.getOrDefault("script", -1).longValue();
+                    if (implicitTimeout != -1)
```
Comment (rzo1) [inline core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java:83]: I would add the `{ }` :-)

Comment (rzo1) [reply core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java:83]: Wonder if we want a (debug) log entry, that the value is unset and the default is used?

**Review by rzo1 — APPROVED** [review]

File: core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java (hunk 2 above)
Comment (jnioche) [reply core/src/main/java/com/digitalpebble/stormcrawler/protocol/selenium/RemoteDriverProtocol.java:83]: > Wonder if we want a (debug) log entry, that the value is unset and the default is used?

we can't tell whether the value has been explicitly set to -1 or if the default set it this way. Let's not bother

**Pull request merged**